typespec-extension/src/code-model-builder.ts

```diff
@@ -1695,275 +1681,201 @@ export class CodeModelBuilder {
+        summary: this.getSummary(type.__raw),
       }),
     );
   }
 
-  private processNumberSchema(type: Scalar, name: string): NumberSchema {
+  private processNumberSchemaFromSdkType(type: SdkBuiltInType, name: string): NumberSchema {
     return this.codeModel.schemas.add(
-      new NumberSchema(name, this.getDoc(type), SchemaType.Number, 64, {
-        summary: this.getSummary(type),
+      new NumberSchema(name, this.getDoc(type.__raw), SchemaType.Number, 64, {
+        summary: this.getSummary(type.__raw),
       }),
     );
   }
 
-  private processDecimalSchema(type: Scalar, name: string): NumberSchema {
+  private processDecimalSchemaFromSdkType(type: SdkBuiltInType, name: string): NumberSchema {
     // "Infinity" maps to "BigDecimal" in Java
     return this.codeModel.schemas.add(
-      new NumberSchema(name, this.getDoc(type), SchemaType.Number, Infinity, {
-        summary: this.getSummary(type),
+      new NumberSchema(name, this.getDoc(type.__raw), SchemaType.Number, Infinity, {
+        summary: this.getSummary(type.__raw),
       }),
     );
   }
 
-  private processBooleanSchema(type: Scalar, name: string): BooleanSchema {
+  private processBooleanSchemaFromSdkType(type: SdkBuiltInType, name: string): BooleanSchema {
     return this.codeModel.schemas.add(
-      new BooleanSchema(name, this.getDoc(type), {
-        summary: this.getSummary(type),
+      new BooleanSchema(name, this.getDoc(type.__raw), {
+        summary: this.getSummary(type.__raw),
       }),
     );
   }
 
-  private processArraySchema(type: ArrayModelType, name: string): ArraySchema {
-    const elementSchema = this.processSchema(type.indexer.value, name);
+  private processArraySchemaFromSdkType(type: SdkArrayType, name: string): ArraySchema {
+    const elementSchema = this.processSchemaFromSdkType(type.valueType, name);
     return this.codeModel.schemas.add(
-      new ArraySchema(name, this.getDoc(type), elementSchema, {
-        summary: this.getSummary(type),
+      new ArraySchema(name, this.getDoc(type.__raw), elementSchema, {
+        summary: this.getSummary(type.__raw),
       }),
     );
   }
 
-  private processDictionarySchema(type: RecordModelType, name: string): DictionarySchema {
-    const dictSchema = new DictionarySchema<any>(name, this.getDoc(type), null, {
-      summary: this.getSummary(type),
+  private processDictionarySchemaFromSdkType(type: SdkDictionaryType, name: string): DictionarySchema {
+    const dictSchema = new DictionarySchema<any>(name, type.details ? type.details : "", null, {
+      summary: type.description ? type.description : "",
     });
 
     // cache this now before we accidentally recurse on this type.
     if (!this.schemaCache.has(type)) {
       this.schemaCache.set(type, dictSchema);
     }
 
-    const elementSchema = this.processSchema(type.indexer.value, name);
+    const elementSchema = this.processSchemaFromSdkType(type.valueType, type.valueType.kind);
```

Comment:
why `type.valueType.kind` as nameHint?

Reply:
No special reason for this, just provide a name hint and not all `SdkType` has `name`, so I use `kind`. I could change back to use `name`.